malsim/sims/mal_simulator.py

```diff
@@ -1,70 +1,40 @@
 from __future__ import annotations
 
-import sys
 import copy
 import logging
-import functools
-from typing import Optional, TYPE_CHECKING
-import numpy as np
-
-from gymnasium.spaces import MultiDiscrete, Box, Dict
-from pettingzoo import ParallelEnv
```

Comment:
All these dependencies were moved to other modules that are all optional to use. The main offering is the mal simulator. These dependencies should also be made [optional to install](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#dependencies-and-requirements) in pyproject.toml. Perhaps behind a `petting-zoo` and/or a `vectors` key(s).